feat: front-end changes to allow gateway renaming

tjvantoll opened the pull request:
# Problem Context

We want to allow users to change the name of their gateways.

## Changes

This PR introduces a new UI on the gateway details page that allows users to update their gateway names inline.

## Screenshot (if applicable)

![gateway-rename](https://user-images.githubusercontent.com/544280/157725297-b75688c4-aa3d-4eeb-8afa-b12bd19e84f7.gif)

## Testing

* You can test this manually by visiting the gateway details page and using the new UI.
* There are also new unit and Cypress tests.

## Any other related PRs

Backend APIs (already merged to `main`): https://github.com/blues/sparrow-reference-web-app/pull/70

## Ticket(s)

https://www.pivotaltracker.com/story/show/181290351

CarltonHenderson commented on src/pages/api/gateway/[gatewayUID]/name.ts:
```diff
@@ -25,7 +25,10 @@ function validateRequest(
 ): false | ValidRequest {
-  const { gatewayUID, name } = req.query;
+  const { gatewayUID } = req.query;
+  // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+  const { name } = req.body;
```
To delete this warning, join the [conversation](https://bluesinc.slack.com/archives/C02LAQLMJR5/p1646952266854589) on data validation.

CarltonHenderson commented on src/styles/Details.module.scss:
```diff
@@ -1,5 +1,13 @@
 @import "_variables.scss";
 
+.editable {
+  display: flex;
+  margin-bottom: 5px;
+  :global .ant-form-inline .ant-form-item {
+    margin-right: 10px;
+  }
+}
+
```
Can we encapsulate the edit-in-place stuff and include this too?

CarltonHenderson reviewed (approved):
Approved w/comments. Well done.

tjvantoll commented on __tests__/src/components/elements/GatewayDetails.test.tsx:
```diff
@@ -0,0 +1,154 @@
+    const viewModel = getGatewayDetailsPresentation(gateway, mockNodes);
+    render(<GatewayDetails viewModel={viewModel} onChangeName={mockChange} />);
+    expect(
+      screen.queryAllByText(GATEWAY_MESSAGE.NO_LOCATION)[0]
```
The logic actually isn’t in the React component anymore, but I never moved this test. I’ll create a `gatewayDetails.test.ts` since that’s where the actual logic is now.

tjvantoll commented on __tests__/src/components/elements/GatewayDetails.test.tsx:
```diff
@@ -0,0 +1,154 @@
+      <GatewayDetails
+        viewModel={viewModel}
+        onChangeName={mockChange}
+        err={errorMessage}
```
It would definitely clean things up. I think I’ll hold off on this PR to avoid touching too much, and if we did that I’d kind of want to do all pages at once.

tjvantoll commented on __tests__/src/components/elements/GatewayDetails.test.tsx:
```diff
@@ -0,0 +1,154 @@
+      />
+    );
+
+    expect(
+      screen.getByText(errorMessage, { exact: false })
+    ).toBeInTheDocument();
+    expect(
+      screen.queryByText("Gateway", { exact: false })
+    ).not.toBeInTheDocument();
+    expect(
+      screen.queryByText("Node", { exact: false })
+    ).not.toBeInTheDocument();
```
Personally I’m ok with the bit of verbosity here.

tjvantoll commented on cypress/integration/homepage.spec.js:
```diff
@@ -153,4 +153,19 @@ describe("Sparrow Application", () => {
+    cy.get("#name").clear();
+    cy.get("#name").type("CYPRESS_TEST");
+    cy.get('[data-testid="gateway-name-submit-button"]').click();
+    cy.get('[data-testid="gateway-details-header"]', { timeout: 5000 }).should(
```
👍

tjvantoll commented:
@CarltonHenderson I think I addressed all your comments. When you have a minute could you take a quick look at `<EditInPlace>`? ([Here’s the code.](https://github.com/blues/sparrow-reference-web-app/blob/tj-gateway-rename-frontend/src/components/elements/EditInPlace.tsx)) Just want a sanity check on the API before I merge this.

CarltonHenderson commented:
LGTM :+1:
